### DISCUSSION TITLE
[Info] How frequently are the checks performed ?

### DISCUSSION THREAD

**Discussion by vinithreddy3** [talsec/Free-RASP-ReactNative#62] [Q&A]
Thanks for this amazing library which includes multiple checks.
As there are listeners added, it should be periodically performing these checks. How frequently are the checks performed and also is there anyway we can configure how frequently are the checks performed ?

After I started using the app I injected Frida script and noticed the **hooks** action listener triggered close to 1min after the script is executed.

**Comment by tompsota** [comment]
Hi @vinithreddy3,

Yes, the checks are performed periodically. The interval is randomized, but no longer than four minutes.
Currently there isn't any option to configure this interval. However, we will think about adding this option in a future release as this could be a useful feature for some apps.

If you have any other question feel free to ask.

Cheers,
Tomas, Talsec developer.

**Comment by vinithreddy3** [comment]
Hi @tompsota, thanks for the quick response. My use case was especially for the hooking detection periodically in intervals of 10-15 secs (or) max 30 secs. Comparatively it looked for me that there aren't many approaches available online to detect runtime hooking frameworks like Frida. Let me look more into that.

**Comment by SWtroy1129** [comment]
Hi would like to ask if this feature is possible in the newer versions? or if there would be any possible workarounds to get the intervals lowered?

**Reply by tompsota** [reply]
Hi @SWtroy1129,

unfortunately, this feature is not currently available in freeRASP, but we have this in our backlog. 

To better understand your needs, can you describe why do you need to perform the checks more often? Do you see any issue with the current approach?
